packages/gated-content/src/conditions/advanced-contract-condition.ts

```diff
@@ -0,0 +1,150 @@
+import { Interface } from '@ethersproject/abi';
+import { BigNumber } from '@ethersproject/bignumber';
+import { AdvancedContractCondition, ConditionComparisonOperator } from '@lens-protocol/metadata';
+import { invariant, assertError } from '@lens-protocol/shared-kernel';
+
+import { LitConditionType, LitEvmAccessCondition } from './types';
+import { toLitSupportedChainName, resolveScalarOperatorSymbol } from './utils';
+import {
+  assertValidAddress,
+  assertSupportedChainId,
+  InvalidAccessCriteriaError,
+} from './validators';
+
+export const transformAdvancedContractCondition = (
+  condition: AdvancedContractCondition,
+): Array<LitEvmAccessCondition> => {
+  assertValidAddress(condition.contract.address);
+  assertSupportedChainId(condition.contract.chainId);
+  assertValidAbi(condition.abi, condition.functionName);
+  assertValidFunctionParams(condition);
+  assertValidComparison(condition);
+
+  return [
+    {
+      conditionType: LitConditionType.EVM_CONTRACT,
+      contractAddress: condition.contract.address.toLowerCase(),
+      chain: toLitSupportedChainName(condition.contract.chainId),
+      functionName: condition.functionName,
+      functionParams: condition.params || [],
+      functionAbi: new Interface([condition.abi]).getFunction(condition.functionName),
+      returnValueTest: {
+        key: '',
+        comparator: resolveScalarOperatorSymbol(condition.comparison),
+        value: condition.value,
+      },
+    },
+  ];
+};
+
+function assertValidAbi(humanReadableAbi: string, functionName: string): void {
+  try {
+    const fn = new Interface([humanReadableAbi]).getFunction(functionName);
+
+    // assert view function
+    invariant(fn.stateMutability === 'view', 'unsupported');
+
+    // assert single output
+    invariant(Array.isArray(fn.outputs) && fn.outputs.length === 1, 'unsupported');
+
+    // assert output is boolean or uint
+    invariant(
+      fn.outputs[0] && (fn.outputs[0].type === 'bool' || fn.outputs[0].type === 'uint256'),
```

Comment:
might break with diff types of uints, suggesting this check instead 
```suggestion
      fn.outputs[0] && (fn.outputs[0].type === 'bool' || /^uint(8|16|32|64|128|256)$/.test(fn.outputs[0].type),
```